provider/queue/queue.go

```diff
@@ -124,7 +80,26 @@ func (q *Queue) work() {
 
 		for {
 			if c == cid.Undef {
-				k, c = q.nextEntry()
+				head, e := q.getQueueHead()
+
+				if e != nil {
+					log.Errorf("error querying for head of queue: %s, stopping provider", e)
+					return
+				} else if head != nil {
+					k = datastore.NewKey(head.Key)
+					c, e = cid.Parse(head.Value)
+					if e != nil {
+						log.Warningf("error parsing queue entry cid with key (%s), removing it from queue: %s", head.Key, e)
+						err := q.ds.Delete(k)
+						if err != nil {
+							log.Errorf("error deleting queue entry with key (%s), due to error (%s), stopping provider", head.Key, err)
+							return
```

Comment:
If we get an error deleting the bad head entry, then the queue will not work so I stop it. Bad idea?